spec.html

```diff
@@ -94,6 +94,69 @@ <h1>ApplyOptionsToTag( tag, options )</h1>
       </emu-alg>
     </emu-clause>
 
+    <emu-clause id="sec-findextension" aoid="FindExtension">
+      <h1>FindExtension ( _extension_, _key_ )</h1>
+
+      <p>The abstract operation FindExtension is called with _extension_, which must be a Unicode locale extension sequence, and String _key_. This operation returns start and end indices for the type subtags for _key_ by performing the following steps:</p>
+
+      <emu-alg>
+        1. Assert: The number of elements in _key_ is 2.
+        1. Let _size_ be the number of elements in _extension_.
+        1. Let _searchValue_ be the concatenation of `"-"`, _key_, and `"-"`.
+        1. Let _pos_ be Call(%StringProto_indexOf%, _extension_, &laquo; _searchValue_ &raquo;).
+        1. If _pos_ ≠ -1, then
+          1. Let _start_ be _pos_ + 4.
+          1. Let _end_ be _start_.
+          1. Let _k_ be _start_.
+          1. Let _done_ be *false*.
+          1. Repeat, while _done_ is *false*
+            1. Let _e_ be Call(%StringProto_indexOf%, _extension_, &laquo; `"-"`, _k_ &raquo;).
+            1. If _e_ = -1, let _len_ be _size_ - _k_; else let _len_ be _e_ - _k_.
+            1. If _len_ = 2, then
+              1. Let _done_ be *true*.
+            1. Else if _e_ = -1, then
+              1. Let _end_ be _size_.
+              1. Let _done_ be *true*.
+            1. Else,
+              1. Let _end_ be _e_.
+              1. Let _k_ be _e_ + 1.
+          1. Return a new Record { [[Start]]: _pos_, [[End]]: _end_ }.
+        1. Let _searchValue_ be the concatenation of `"-"` and _key_.
+        1. Let _pos_ be Call(%StringProto_indexOf%, _extension_, &laquo; _searchValue_ &raquo;).
+        1. If _pos_ ≠ -1 and _pos_ + 3 = _size_, then
+          1. Return a new Record {{ [[Start]]: _pos_, [[End]]: _size_ }.
+        1. Return *undefined*.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-unicodeextensionvalue" aoid="UnicodeExtensionValue">
+      <h1>UnicodeExtensionValue ( _extension_, _key_ )</h1>
+
+      <p>The abstract operation UnicodeExtensionValue is called with _extension_, which must be a Unicode locale extension sequence, and String _key_. This operation returns the type subtags for _key_ by performing the following steps:</p>
+
+      <emu-alg>
+        1. Let _range_ be FindExtension(_extension_, _key_).
+        1. If _range_ is *undefined*, return *undefined*.
+        1. If _range_.[[Start]] + 4 &ge; _range_.[[End]], return an empty String.
+        1. Otherwise, return a String value equal to the substring of _extension_ from _range_.[[Start]] + 4 (inclusive) to _range_.[[End]] (exclusive).
```

Comment:
nit: is it clearer here to say `Call(%StringProto_substring%, _extension_, _range_.[[Start]] + 4, _range_.[[End]])` ?

Reply:
I don't see why it would be clearer. It seems like the current ECMA 262 and ECMA 402 specifications use this sort of wording. Do you prefer the alternative? If so, maybe we should make this as a separate editorial change against the whole specification.

Reply:
It would encode the inclusivity/exclusivity and would be slightly less wordy. I don't have strong feelings on it though, theres already a mix of descriptive psuedo-code like this with exact std method calls throughout.